packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -238,22 +270,64 @@ class AppSizeController {
     changeAnalysisJsonFile(jsonFile);
 
     // Set deferred app flag.
-    isDeferredApp =
-        deferredLoadingSupportEnabled && processedJson['n'] == 'ArtificialRoot';
+    _isDeferredApp.value =
+        deferredLoadingSupportEnabled && _hasDeferredInfo(processedJson);
+
+    if (isDeferredApp.value) {
+      _deferredOnly = _extractDeferredUnits({...processedJson});
+      _mainOnly = _extractMainUnit({...processedJson});
+      _entireApp = _includeEntireApp({...processedJson});
+      _loadApp(_activeAppUnit!);
+    } else {
+      _loadApp(processedJson);
+    }
+
+    _processingNotifier.value = false;
+  }
 
-    // Set root name.
-    processedJson['n'] = isDeferredApp ? 'Entire app' : 'Root';
```

Comment:
do we still need to set `processedJson['n']` to 'Root' somewhere for non-deferred apps?

Reply:
Yes :) The test caught this too lol